.github/workflows/iroha2-dev-pr.yml

```
@@ -144,44 +150,67 @@ jobs:
           name: clippy.json
           path: clippy.json
 
-  torii-api-and-client-cli-tests:
+  torii-api-client-cli-docker-compose-tests:
     runs-on: [self-hosted, Linux, iroha2]
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-04-18
     timeout-minutes: 60
     steps:
-      - uses: actions/checkout@v4
-      - uses: Swatinem/rust-cache@v2
-      - name: Build binaries
-        run: |
-          cargo build -p iroha_client_cli -p kagami -p irohad
-      - name: Setup test Iroha 2 environment on the bare metal
-        run: |
-          pip3 install -r scripts/requirements.txt --no-input --break-system-packages
-          ./scripts/test_env.py setup
-      - name: Mark binaries as executable
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        id: buildx
+        if: always()
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+      - name: Install Python and Poetry
         run: |
-          chmod +x ${{ env.CLIENT_CLI_DIR }}
-      - name: Install torii api dependencies using Poetry
+          yum install -y python${{ env.PYTHON_VERSION }} python${{ env.PYTHON_VERSION }}-devel
+          curl -sSL https://install.python-poetry.org | python3 -
+          echo 'export PATH="${{ env.POETRY_PATH }}:$PATH"' >> /etc/profile
+          source /etc/profile
+      - name: Install Torii API Dependencies
         working-directory: torii/pytests
+        run: ${{ env.POETRY_PATH }} install
+      - name: Build and Tag Docker Images
+        uses: docker/build-push-action@v6
+        if: always()
+        with:
+          context: .
+          load: true
+          file: Dockerfile
+          tags: |
+            hyperledger/iroha:local
+            hyperledger/iroha:dev
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Test docker-compose.single.yml
+        run: |
+          docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.single.yml up --wait || exit 1
```

```suggestion
          docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.single.yml up --wait
```

doesn't docker-compose return error code other than 0? why do you insist on returning `1`?

---

yeah, I was thinking that `docker-compose` returns an error code on unhealty status

---

It could be unhealthy status but it doesn't mean that Actions job will be exited